<?xml version="1.0"?>
<launch>
  <arg name="headless" default="false" />
  <arg name="image_name" default="/camera/image" />
  <arg name="plugin_name" default="ConditionalHsvFilter" />

  <!-- color region detector -->
  <node pkg="jsk_uav_forest_perception" type="vision_detection_node" name="vision_detection"
        output="screen">
    <param name="image_topic_name" value="$(arg image_name)"/>
    <param name="camera_info_topic_name" value="/camera/camera_info" />
    <param name="laser_scan_topic_name" value="/scan_clustered"/>
    <param name="verbose" value="true"/>
    <param name="detect_once" value="true"/>
    <param name="camera_offset_x" value="0.0" />
    <param name="first_detection_depth_thre" value="12.0" /> <!-- should be 9[m] -->
    <param name="plugin_name" value="vision_detection/$(arg plugin_name)" />
    <rosparam file="$(find jsk_uav_forest_perception)/param/$(arg plugin_name).yaml"/>
  </node>

</launch>
